cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,105 @@
+from ast import alias
+import os
+import json
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        if cloud == 'aws':
+            self.cloud_instance = 'aws_instance'
+        self.resources = {'resource': {self.cloud_instance: {}}}
+        self.providers = {'provider': {cloud: []}}
+
+    def __del__(self):
+        self.destroy_infra()
+
+    def __dump_to_json(self, content, file):
+        with open(file, 'w') as config_file:
+            json.dump(content, config_file)
+
+    def set_aws_profile(self, profile):
+        self.aws_profile = profile
+
+    def add_region(self, region):
+        if self.cloud == 'aws':
+            # If no profile was specified, select the default
+            if not hasattr(self, 'aws_profile'):
+                self.aws_profile = 'aws'
+
+            new_provider = {'region': region, 'alias': region, 'profile': self.aws_profile}
+
+        self.providers['provider'][self.cloud].append(new_provider)
+
+    def add_instance(self, region, image_id, instance_type=''):
+        aliases = [provider["alias"] for provider in self.providers["provider"][self.cloud]]
+        if not region in aliases:
+            print('Cannot add an instance if region provider is not set up')
+            exit(1)
+
+        if self.cloud == 'aws':
+            if not instance_type:
+                instance_type = 't2.micro'
+
+            instance_name = f'{region}-{instance_type}'.replace('.', '-')
+            new_instance = {
+                'instance_type': instance_type,
+                'ami': image_id,
+                'provider': f'aws.{region}',
+                'tags': {'name': instance_name},
+            }
+            self.resources['resource']['aws_instance'][instance_name] = new_instance
+            return instance_name
+
+    def get_instances(self):
+        if self.cloud == 'aws':
+            output = os.popen('terraform show --json').read()
+            json_output = json.loads(output)
+
+            resources = json_output['values']['root_module']['resources']
+
+            instances_info = {}
+            for resource in resources:
+                instances_info[resource['address']] = {
+                    'instance_id': resource['values']['id'],
+                    'public_ip': resource['values']['public_ip'],
```

Comment:
Can we add the following fields?:
- public_dns
- availability_zone
- ami

(I will try to use that data to build a report summary).